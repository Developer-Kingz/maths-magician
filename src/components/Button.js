import React from 'react';
import '../App.css';
import PropTypes from 'prop-types';

const Button = ({ text, className }) => (
  <div>
    <button type="button" className={className}>
      {text}
    </button>
  </div>
);

Button.propTypes = {
  text: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};
export default Button;
