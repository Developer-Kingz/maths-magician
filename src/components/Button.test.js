import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Button from "./Button";
import "@testing-library/jest-dom/extend-expect";

describe("Button", () => {
    it("renders button text correctly", () => {
        const buttonText = "Click Me";
        const { getByText } = render(<Button text={buttonText} />);
        const buttonElement = getByText(buttonText);
        expect(buttonElement).toBeInTheDocument();
    });

    it("triggers onClick function when clicked", () => {
        const onClickMock = jest.fn();
        const { getByText } = render(
            <Button text="Click Me" onClick={onClickMock} />
        );
        const buttonElement = getByText("Click Me");
        fireEvent.click(buttonElement);
        expect(onClickMock).toHaveBeenCalled();
    });

    it("applies className correctly", () => {
        const { container } = render(
            <Button text="Click Me" className="custom-button" />
        );
        const buttonElement = container.querySelector("button");
        expect(buttonElement).toHaveClass("custom-button");
    });

    // Add more test cases here as needed.
});
