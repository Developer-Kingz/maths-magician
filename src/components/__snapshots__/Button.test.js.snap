// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      class="custom-btn"
      type="button"
    >
      Click Me
    </button>
  </div>
</DocumentFragment>
`;
